### PR TITLE
do not try to claim disk that doesn't belong to Ceph

### PR DESCRIPTION
**Pull request by jsuchome**
Based on https://github.com/crowbar/barclamp-crowbar/pull/1306 and https://github.com/crowbar/barclamp-deployer/pull/248

### CODE REVIEW

File: chef/cookbooks/ceph/recipes/osd.rb [hunk 1]
```diff
@@ -69,7 +69,15 @@
   if is_crowbar?
     node.set["ceph"]["osd_devices"] = [] if node["ceph"]["osd_devices"].nil?
     min_size_blocks = node["ceph"]["osd"]["min_size_gb"] * 1024 * 1024 * 2
-    unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node).sort.select {|d| d.size >= min_size_blocks}
+
+    claim_string = "Ceph"
```
Comment (tboerger) [inline chef/cookbooks/ceph/recipes/osd.rb]: I would prefer snakecase claim strings as we anyway display some I18n key within crowbar.

Comment (tboerger) [inline chef/cookbooks/ceph/recipes/osd.rb]: Ok, forget it. we are already using this name.

**Comment by tboerger** [comment]
LGTM

File: chef/cookbooks/ceph/recipes/osd.rb [hunk 2]
```diff
@@ -69,7 +69,15 @@
...
+
+    # do not try to claim disk that belongs to something non-Ceph
+    disk_roles = node["crowbar_wall"].disk_roles || {}
+    unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node).sort.select {|d|
+      role = disk_roles[d.unique_name] || ""
```
Comment (jsuchome) [inline chef/cookbooks/ceph/recipes/osd.rb]: see the answer on cinder

**Comment by vuntz** [comment]
+1, but I understand code might change after discussion in https://github.com/crowbar/barclamp-deployer/pull/248

**Comment by rsalevsky** [comment]
@jsuchome, this repository got merged into [crowbar-ceph](https://github.com/crowbar/crowbar-ceph).
Please open a new pull request there. __Do NOT merge here anymore!__

**Comment by jsuchome** [comment]
Closing, as there's already PR to new repo

**Pull request closed without merging**